### PR TITLE
Submit/pippete

### PR DESCRIPTION
**Pull request by Dimava**
- [x] works with deselect keybind
- [x] selects building
- [x] selects variant
- [x] selects rotation
- [x] selects belts
- [x] selects undergrounds
- [x] make sure it does not crash on every one of entities
- [x] select miner on ores

...
- [x] __REMAKE WITH LESS *CODE__

### CODE REVIEW

**Comment by Dimava** [comment]
Everything is working, ready to merge

**Comment by bbugh** [comment]
Looks awesome, can't wait to play with it.

PS it's spelled "pipette". The code has "pippete" in most places.

File: src/js/game/hud/parts/building_placer.js [hunk 1]
```diff
@@ -105,6 +107,49 @@ export class HUDBuildingPlacer extends BaseHUDPart {
...
+                this.currentMetaBuilding.set(gMetaBuildingRegistry.findById("belt"));
+            }
+            return;
+        }
+
+        let match = staticEntity.spriteKey.match(
+            /sprites\/buildings\/(.*?)(_entry|_exit|_[a-z]+|)(|-(.*))\.png/
+        );
+        const id = match[1];
+        const variant = match[4] || defaultBuildingVariant;
```
Comment (Dimava) [inline src/js/game/hud/parts/building_placer.js:145]: If I understood correctly, buildings do not store their ids and variants

Comment (tobspr) [reply src/js/game/hud/parts/building_placer.js:145]: Oof, this is a pretty hacky solution. I'd prefer to find a better solution before mergin this

**Comment by tobspr** [comment]
Hm, this is too hacky - I need to refactor the entity components first so they store information about the variant and such

**Comment by tobspr** [comment]
Kinda struggling with this PR since a lot of people requested it already. I'll keep this PR open as a reminder

**Comment by pfargo** [comment]
Don't code in JS but  pipette functionality is simply a mousedown  at the cursor location and then the "C" key function from the blueprint. Am I missing something?

**Comment by tobspr** [comment]
This will be in 1.1.16. I reused some parts of the code and rewrote others. Thanks a lot!

**Pull request closed without merging**